python-checks/src/test/java/org/sonar/python/checks/quickfix/PythonQuickFixVerifier.java

```diff
@@ -0,0 +1,125 @@
+import org.sonar.plugins.python.api.tree.FileInput;
+import org.sonar.python.SubscriptionVisitor;
+import org.sonar.python.parser.PythonParser;
+import org.sonar.python.quickfix.IssueWithQuickFix;
+import org.sonar.python.semantic.ProjectLevelSymbolTable;
+import org.sonar.python.tree.PythonTreeMaker;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PythonQuickFixVerifier {
+  private PythonQuickFixVerifier() {
+  }
+
+  public static void verifyNoQuickFix(PythonCheck check, String code){
+    List<PythonCheck.PreciseIssue> issuesWithQuickFix = PythonQuickFixVerifier.getIssuesWithQuickFix(check, code);
+    assertThat(issuesWithQuickFix).isEmpty();
+  }
+
+  public static void verify(PythonCheck check, String codeWithIssue, String codeFixed) {
+    List<PythonCheck.PreciseIssue> issues = PythonQuickFixVerifier
+      .getIssuesWithQuickFix(check, codeWithIssue);
+
+    assertThat(issues).hasSize(1);
+    IssueWithQuickFix issue = (IssueWithQuickFix) issues.get(0);
+
+    assertThat(issue.getQuickFixes()).hasSize(1);
+
+    String codeQFApplied = PythonQuickFixVerifier.applyQuickFix(codeWithIssue, issue);
+    assertThat(codeQFApplied).isEqualTo(codeFixed);
+  }
+
+  private static List<PreciseIssue> scanFileForIssues(PythonCheck check, PythonVisitorContext context) {
+    check.scanFile(context);
+    if (check instanceof PythonSubscriptionCheck) {
+      SubscriptionVisitor.analyze(Collections.singletonList((PythonSubscriptionCheck) check), context);
+    }
+    return context.getIssues();
+  }
+
+  public static List<PreciseIssue> getIssuesWithQuickFix(PythonCheck check,String codeWithIssue) {
+    PythonParser parser = PythonParser.create();
+    PythonQuickFixFile pythonFile = new PythonQuickFixFile(codeWithIssue);
+    AstNode astNode = parser.parse(pythonFile.content());
+    FileInput parse = new PythonTreeMaker().fileInput(astNode);
+
+    PythonVisitorContext visitorContext = new PythonVisitorContext(parse,
+      pythonFile, null, "",
+      ProjectLevelSymbolTable.empty());
+
+    return scanFileForIssues(check, visitorContext);
+  }
+
+  public static String applyQuickFix(String codeWithIssue, IssueWithQuickFix issueWithQuickFix) {
+    IssueLocation loc = issueWithQuickFix.getQuickFixes().get(0).getTextEdits().get(0).issueLocation;
+    String replacement = loc.message();
+    int start = convertPositionToIndex(codeWithIssue, loc.startLine(), loc.startLineOffset());
+    int end = convertPositionToIndex(codeWithIssue, loc.endLine(), loc.endLineOffset());
+    return codeWithIssue.substring(0, start) + replacement + codeWithIssue.substring(end);
+  }
+
+  private static int convertPositionToIndex(String fileContent, int line, int lineOffset) {
+    int currentLine = 1;
+    int currentIndex = -1;
+    while (currentLine < line) {
+      currentIndex = fileContent.indexOf("\n", currentIndex + 1);
+      currentLine++;
+    }
+    return currentIndex + lineOffset + 1;
```

Comment:
To me, it would be more understandable like that:
```suggestion
    int currentLine = 1;
    int currentIndex = 0;
    while (currentLine < line) {
      currentIndex = fileContent.indexOf("\n", currentIndex) + 1;
      currentLine++;
    }
    return currentIndex + lineOffset;
```
However, I'm not sure that it would be correct...